lib/OpenQA/Task/Job/Limit.pm

```diff
@@ -24,18 +24,27 @@ sub register {
     $app->minion->add_task(limit_results_and_logs => sub { _limit($app, @_) });
 }
 
+sub _limit_group {
+    my $group            = shift;
+    my $important_builds = $group->important_builds;
+    for my $job (@{$group->find_jobs_with_expired_results($important_builds)}) {
+        $job->delete;
+    }
+    for my $job (@{$group->find_jobs_with_expired_logs($important_builds)}) {
+        $job->delete_logs;
+    }
+
+}
+
 sub _limit {
     my ($app, $job) = @_;
 
+    my $homeless = $app->db->resultset('JobGroups')->new({});
```

Comment:
I assume this is only done to use the methods and defaults available in `JobGroups.pm` for groupless jobs, too? Maybe a comment describing *why* a new group is created here wouldn't hurt. And it should also state that this won't actually create the group in the database which is not that obvious.